src/pczt/tx_extractor.rs

```diff
@@ -144,18 +146,22 @@ impl<V> crate::Bundle<Unbound, V> {
         sighash: [u8; 32],
         rng: R,
     ) -> Option<crate::Bundle<Authorized, V>> {
-        if self
-            .shielded_spends()
-            .iter()
-            .all(|spend| spend.rk().verify(&sighash, spend.spend_auth_sig()).is_ok())
-        {
+        if self.shielded_spends().iter().all(|spend| {
+            spend
+                .rk()
+                .verify(&sighash, spend.spend_auth_sig().sig())
```

Comment:
`spend.spend_auth_sig().sig()` looks repetative. Let's rename to `spend.spend_auth.sig()` and 
```rust
pub struct SpendDescription<A: Authorization> {
    cv: ValueCommitment,
    anchor: bls12_381::Scalar,
    nullifier: Nullifier,
    rk: redjubjub::VerificationKey<SpendAuth>,
    zkproof: A::SpendProof,
    spend_auth: A::AuthSig, // remove _sig here
}
```